components/fxa-client/src/http_client.rs

```diff
@@ -661,14 +662,31 @@ pub struct DeviceResponseCommon {
     pub push_endpoint_expired: bool,
 }
 
+// We modele the OAuthTokenRequest according to the up to date
+// definition on
+// https://github.com/mozilla/fxa/blob/8ae0e6876a50c7f386a9ec5b6df9ebb54ccdf1b5/packages/fxa-auth-server/lib/oauth/routes/token.js#L70-L152
+
 #[derive(Serialize)]
-pub struct OAuthTokenRequest {
-    pub client_id: String,
-    pub grant_type: String,
-    pub refresh_token: String,
-    pub scope: String,
-    #[serde(skip_serializing_if = "Option::is_none")]
-    pub ttl: Option<u64>,
+#[serde(untagged)]
+enum OAauthTokenRequest {
+    RefreshToken {
```

Comment:
Maybe to make it clear, we could rename these variants `UsingRefreshToken`, `UsingCode`?

Reply:
Yes seems a good name. I was using `WithRefreshToken` but the syntax checker advised me not to